pnpbridge/src/adapters/src/impinj_reader_r700/impinj_reader_r700.c

```diff
@@ -0,0 +1,666 @@
+static const int commandStatusFailure = 500;
+
+//
+// Response to various commands [Must be valid JSON]
+//
+// static const unsigned char sampleEnviromentalSensor_BlinkResponse[] = "{ \"status\": 12, \"description\": \"leds blinking\" }";
+// static const unsigned char sampleEnviromentalSensor_TurnOnLightResponse[] = "{ \"status\": 1, \"description\": \"light on\" }";
+// static const unsigned char sampleEnviromentalSensor_TurnOffLightResponse[] = "{ \"status\": 1, \"description\": \"light off\" }";
+static const unsigned char impinjReader_EmptyBody[] = "\" \"";
+
+static const unsigned char impinjReader_OutOfMemory[] = "\"Out of memory\"";
+static const unsigned char impinjReader_NotImplemented[] = "\"Requested command not implemented on this interface\"";
+
+//
+// Property names that are updatable from the server application/operator
+//
+static const char impinjReader_property_testPropertyString[] = "testPropertyString";
+
+//
+// Response description is an optional, human readable message including more information
+// about the setting of the property
+//
+static const char g_ImpinjReaderPropertyResponseDescription[] = "success";
+
+// Format of the body when responding to a targetTemperature 
+// static const char g_environmentalSensorBrightnessResponseFormat[] = "%.2d";
+
+
+// ImpinjReader_SetCommandResponse is a helper that fills out a command response
+static int ImpinjReader_SetCommandResponse(
+    unsigned char** CommandResponse,
+    size_t* CommandResponseSize,
+    const unsigned char* ResponseData)
+{
+    int result = PNP_STATUS_SUCCESS;
+    if (ResponseData == NULL)
+    {
+        LogError("Impinj Reader Adapter:: Response Data is empty");
+        *CommandResponseSize = 0;
+        return PNP_STATUS_INTERNAL_ERROR;
+    }
+
+    *CommandResponseSize = strlen((char*)ResponseData);
+    memset(CommandResponse, 0, sizeof(*CommandResponse));
+
+    // Allocate a copy of the response data to return to the invoker. Caller will free this.
+    if (mallocAndStrcpy_s((char**)CommandResponse, (char*)ResponseData) != 0)
+    {
+        LogError("Impinj Reader Adapter:: Unable to allocate response data");
+        result = PNP_STATUS_INTERNAL_ERROR;
+    }
+
+    return result;
+}
+
+static void ImpinjReader_PropertyCallback(
+        int ReportedStatus, 
+        void* UserContextCallback)
+        {
+            LogInfo("PropertyCallback called, result=%d, property name=%s", ReportedStatus, (const char*)UserContextCallback);
+        }
+
+IOTHUB_CLIENT_RESULT ImpinjReader_ReportDeviceStateAsync(
+    PNPBRIDGE_COMPONENT_HANDLE PnpComponentHandle,
+    const char * ComponentName)
+    {
+        PIMPINJ_READER device = PnpComponentHandleGetContext(PnpComponentHandle);
+        char * result = curlStaticGet(device->curl_static_session, "/status");
+        return ImpinjReader_ReportPropertyAsync(PnpComponentHandle, ComponentName, "deviceStatus", result);
+    }
+
+char * ImpinjReader_CreateJsonResponse(
```

Comment:
>ImpinjReader_CreateJsonResponse [](start = 7, length = 31)

Can't we do this with sprintf and a message format?

Reply:
Yes i believe we can!  fixed in - a96d8a2: replaced strcat with sprintf commands